api/lib/infrastructure/airtable.js

```diff
@@ -1,76 +1,42 @@
 const Airtable = require('airtable');
-const airtableConfig = require('../settings').airtable;
 const AirtableRecord = Airtable.Record;
-const cache = require('./caches/cache');
-const logger = require('./logger');
+const airtableConfig = require('../settings').airtable;
+const airtableCache = require('./airtable-cache');
 
 module.exports = {
 
-  _base() {
-    return new Airtable({ apiKey: airtableConfig.apiKey }).base(airtableConfig.base);
+  async getRecord(tableName, recordId) {
+    const cachedRawJson = await airtableCache.get(tableName, recordId, this._getAirtableRecord(tableName, recordId));
+
+    return new AirtableRecord(tableName, cachedRawJson.id, cachedRawJson);
   },
 
-  getRecord(tableName, recordId) {
-    const cacheKey = `${tableName}_${recordId}`;
-    const logContext = {
-      zone: 'airtable.getRecord',
-      type: 'airtable',
-      recordId,
-      tableName,
-    };
+  async findRecords(tableName) {
+    const cachedArrayOfRawJson = await airtableCache.get(tableName, null, this._findAirtableRecords(tableName));
 
-    return cache.get(cacheKey)
-      .then((cachedRawJson) => {
-        if (cachedRawJson) {
-          return new AirtableRecord(tableName, cachedRawJson.id, cachedRawJson);
-        }
+    return cachedArrayOfRawJson.map((rawJson) => new AirtableRecord(tableName, rawJson.id, rawJson));
+  },
 
-        logger.trace(logContext, 'cache miss. Calling airtable');
-        return this._base()
-          .table(tableName)
-          .find(recordId)
-          .then((record) => {
-            logger.trace(logContext, 'found record in Airtable.');
-            return cache.set(cacheKey, record._rawJson)
-              .then(() => record);
-          })
-          .catch((err) => {
-            logContext.err = err;
-            logger.error(logContext, 'Airtable error');
-            return Promise.reject(err);
-          });
-      });
+  _airtableClient() {
```

Comment:
Il vaudrait mieux remonter toutes les fonctions privées (ou publiques mais appelées en interne dans le module) au niveau du module (plutôt que dans `exports`) plutôt qu'utiliser des `this.` qui sont fragiles.